### PR TITLE
move deploy to deploy stage

### PR DESCRIPTION
**Pull request by minrk**
avoids restricting testing to deployable branches

### CODE REVIEW

**Comment by minrk** [comment]
Needs to wait for #325 and rebase, since it changes how build.py is called slightly.

Will need to double check that the deploy succeeded after merging this one!

**Comment by minrk** [comment]
Rebased on #330, so if that passes, this should be okay.

File: .travis.yml [hunk 1]
```diff
@@ -8,19 +8,17 @@ install:
 - . ci/minikube.env
 - ./ci/install.sh
 script:
-- |
-  if [[ "$TRAVIS_PULL_REQUEST" == "false" && ( "$TRAVIS_BRANCH" = master || "$TRAVIS_BRANCH" == v0.4.x ) ]]; then
-    docker login -u ${DOCKER_USERNAME} -p "${DOCKER_PASSWORD}"
-    export PUSH="--push --publish-chart"
-  fi
-- "GIT_SSH_COMMAND='ssh -i travis' ./build.py --commit-range ${TRAVIS_COMMIT_RANGE} $PUSH"
+- ./build.py build --commit-range ${TRAVIS_COMMIT_RANGE}
```
Comment (yuvipanda) [inline .travis.yml]: need to get rid of the second 'build', that's why this is failing

**Comment by yuvipanda** [comment]
Thanks, @minrk

**Pull request merged**